firmware-support/memmap-generate/src/parse.rs

```diff
@@ -0,0 +1,567 @@
+                "file": "src/Protocols/MemoryMap.hs",
+                "module": "Protocols.MemoryMap",
+                "package": "clash-protocols-memmap-0.1-inplace",
+                "start_col": 15,
+                "start_line": 435
+              }
+            }
+          }
+        },
+        {
+          "relative_address": 2684354560,
+          "size": 536870911,
+          "tree": {
+            "device_instance": {
+              "absolute_address": 2684354560,
+              "device_name": "Timer",
+              "instance_name": "Timer",
+              "src_location": {
+                "end_col": 24,
+                "end_line": 435,
+                "file": "src/Protocols/MemoryMap.hs",
+                "module": "Protocols.MemoryMap",
+                "package": "clash-protocols-memmap-0.1-inplace",
+                "start_col": 15,
+                "start_line": 435
+              }
+            }
+          }
+        },
+        {
+          "relative_address": 3221225472,
+          "size": 536870911,
+          "tree": {
+            "device_instance": {
+              "absolute_address": 3221225472,
+              "device_name": "UART",
+              "instance_name": "UART",
+              "src_location": {
+                "end_col": 60,
+                "end_line": 45,
+                "file": "src/Bittide/Instances/MemoryMapLogic.hs",
+                "module": "Bittide.Instances.MemoryMapLogic",
+                "package": "bittide-instances-0.1-inplace",
+                "start_col": 50,
+                "start_line": 45
+              }
+            }
+          }
+        },
+        {
+          "relative_address": 3758096384,
+          "size": 536870911,
+          "tree": {
+            "device_instance": {
+              "absolute_address": 3758096384,
+              "device_name": "StatusReg",
+              "instance_name": "StatusReg",
+              "src_location": {
+                "end_col": 24,
+                "end_line": 435,
+                "file": "src/Protocols/MemoryMap.hs",
+                "module": "Protocols.MemoryMap",
+                "package": "clash-protocols-memmap-0.1-inplace",
+                "start_col": 15,
+                "start_line": 435
+              }
+            }
+          }
+        }
+      ],
+      "src_location": {
+        "end_col": 48,
+        "end_line": 233,
+        "file": "src/Bittide/ProcessingElement.hs",
+        "module": "Bittide.ProcessingElement",
+        "package": "bittide-0.1-inplace",
+        "start_col": 23,
+        "start_line": 233
+      }
+    }
+  }
+}
+
+"#;
+
+    #[test]
+    fn register_access() {
+        let src = "\"read_only\"";
+        let x: RegisterAccess = serde_json::from_str(src).unwrap();
+        assert_eq!(x, RegisterAccess::ReadOnly);
+    }
+
+    #[test]
+    fn test_deserialise_memmap() {
+        let memmap: MemoryMapDesc = serde_json::from_str(TEST_SRC).unwrap();
+
+        dbg!(&memmap);
+    }
+}
```

Comment:
Could we use an actual generated description from a clash design instead of hardcoding this?